### PR TITLE
chore(enzyme replacement in test files) : User Story 2132089

### PR DESCRIPTION
**Pull request by v-sharmachir**
#### Details

Replacement of Enzyme in files under src/tests/unit/tests/common/component

##### Motivation
Existing story: [User Story: 2132089](https://dev.azure.com/mseng/1ES/_workitems/edit/2132089)

Replaced the use of Enzyme in the below files:

all files in controls sub folder
all files in scanning-spinner sub folder
blocking-dialog.test.tsx
body-class-modifier.test.tsx
collapsible-component.test.tsx
copy-issue-details-button.test.tsx
enable-telemetry-setting-description.test.tsx
expand-collapse-left-nav-hamburger-button.test.tsx
external-link.test.tsx
flagged-component.test.tsx
header-icon.test.tsx
header.test.tsx
heading-with-content-link.test.tsx
issue-filing-button.test.tsx
issue-filing-needs-settings-help-text.test.tsx

Code coverage before and after codeshift checked and after codeshift it is not reduced.

#### Pull request checklist

- [x] Addresses an existing issue: [User Story: 2132089](https://dev.azure.com/mseng/1ES/_workitems/edit/2132089)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/common/components/copy-issue-details-button.test.tsx [hunk 1]
```diff
@@ -66,16 +76,14 @@ describe('CopyIssueDetailsButtonTest', () => {
...
 
-            const result = Enzyme.mount(<CopyIssueDetailsButton {...props} />);
-            const button = result.find(DefaultButton);
+            const result = render(<CopyIssueDetailsButton {...props} />);
             onClickMock.setup(m => m(It.isAny())).verifiable(Times.once());
             // tslint:disable-next-line: await-promise
-            await button.simulate('click');
-
-            const toast = result.find(Toast);
+            await userEvent.click(result.container.querySelector('.ms-Button-label'));
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/copy-issue-details-button.test.tsx:82]: For this one, it's worth it to not mock `DefaultButton` and use `result.getByRole('button')` for clarity.

File: src/tests/unit/tests/common/components/issue-filing-button.test.tsx [hunk 2]
```diff
@@ -98,20 +110,22 @@ describe('IssueFilingButtonTest', () => {
...
-            .setup(creator =>
-                creator.fileIssue(eventStub, testKey, props.issueDetailsData, toolData),
-            )
+            .setup(creator => creator.fileIssue(It.isAny(), It.isAny(), It.isAny(), It.isAny()))
             .verifiable(Times.once());
-        const wrapper = shallow<IssueFilingButton>(<IssueFilingButton {...props} />);
+        const renderResult = render(<IssueFilingButton {...props} />);
 
-        wrapper.find(DefaultButton).simulate('click', eventStub);
+        await userEvent.click(renderResult.container.querySelector('.ms-Button-label'));
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/issue-filing-button.test.tsx:117]: Another instance where it's better to *not* mock `DefaultButton` so we can use `getByRole('button')`

File: src/tests/unit/tests/common/components/scanning-spinner/__snapshots__/scanning-spinner.test.tsx.snap [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
 exports[`ScanningSpinner renders isScanning = <true> 1`] = `
-<StyledSpinnerBase
-  aria-live="test-aria-live"
-  className="scanningSpinner"
-  label="test-label"
-  role="alert"
-  size={3}
-/>
+<DocumentFragment>
+  <div
```
Comment (madalynrose) [inline src/tests/unit/tests/common/components/scanning-spinner/__snapshots__/scanning-spinner.test.tsx.snap:7]: It'd be good to mock `Spinner` so this snapshot is closer to the old one.

**Review by madalynrose — CHANGES_REQUESTED** [review]
A few small adjustments to mocked components and selector calls but this looks great overall!

File: src/tests/unit/tests/common/components/__snapshots__/collapsible-component.test.tsx.snap [hunk 4]
```diff
@@ -1,162 +1,210 @@
...
-    </div>
-  </CustomizedActionButton>
+<DocumentFragment>
   <div
-    className="content-class-name collapsibleContent"
+    class=""
   >
-    <div>
-      Some content
+    <button
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/__snapshots__/collapsible-component.test.tsx.snap]: resolved the comment

File: src/tests/unit/tests/common/components/__snapshots__/expand-collapse-left-nav-hamburger-button.test.tsx.snap [hunk 5]
```diff
@@ -1,28 +1,73 @@
...
 
 exports[`AssessmentLeftNavHamburgerButton renders per snapshot 1`] = `
-<LeftNavHamburgerButton
-  ariaLabel="Assessment - all tests and requirements list"
-  className="some-class"
-  isSideNavOpen={false}
-  setSideNavOpen={null}
-/>
+<DocumentFragment>
+  <button
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/__snapshots__/expand-collapse-left-nav-hamburger-button.test.tsx.snap]: resolved the comment

File: src/tests/unit/tests/common/components/__snapshots__/heading-with-content-link.test.tsx.snap [hunk 6]
```diff
@@ -1,34 +1,114 @@
...
+    <mock-contentlink
+      deps="[object Object]"
+      iconname="info"
+      reference="for/testing"
+    />
+  </div>
+</DocumentFragment>
+`;
+
+exports[`HeadingWithContentLink renders with both text and icon: ContentLink props 1`] = `
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/__snapshots__/heading-with-content-link.test.tsx.snap]: Removed the extra expanded deps to make snapshot similar to old version

File: src/tests/unit/tests/common/components/__snapshots__/issue-filing-button.test.tsx.snap [hunk 7]
```diff
@@ -1,49 +1,101 @@
...
+    isopen="true"
+    issuefilingservicepropertiesmap="[object Object]"
+    selectedissuedata="[object Object]"
+    selectedissuefilingservice="[object Object]"
+    selectedissuefilingservicedata="[object Object]"
+  />
+</DocumentFragment>
 `;
 
+exports[`IssueFilingButtonTest onclick: invalid settings, %s: LadyBugSolidIcon props 1`] = `{}`;
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/__snapshots__/issue-filing-button.test.tsx.snap]: removed the unnecessary prop

File: src/tests/unit/tests/common/components/controls/__snapshots__/insights-command-button.test.tsx.snap [hunk 8]
```diff
@@ -1,47 +1,72 @@
...
-<CustomizedActionButton
-  className="insightsCommandButton test-extra-class-name"
-  iconProps={
-    {
-      "className": "commandBarButtonIcon",
-    }
-  }
-/>
+<DocumentFragment>
+  <button
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/controls/__snapshots__/insights-command-button.test.tsx.snap]: resolved the comment

File: src/tests/unit/tests/common/components/copy-issue-details-button.test.tsx (hunk 1 above)
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/copy-issue-details-button.test.tsx]: resolved the comment

File: src/tests/unit/tests/common/components/issue-filing-button.test.tsx (hunk 2 above)
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/issue-filing-button.test.tsx]: resolved the comment

File: src/tests/unit/tests/common/components/scanning-spinner/__snapshots__/scanning-spinner.test.tsx.snap (hunk 3 above)
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/scanning-spinner/__snapshots__/scanning-spinner.test.tsx.snap]: resolved the comment

**Review by madalynrose — CHANGES_REQUESTED** [review]
A few small tweaks and then this looks awesome!

File: src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx [hunk 9]
```diff
@@ -1,51 +1,61 @@
...
             <FastPassLeftNavHamburgerButton
                 isSideNavOpen={false}
                 setSideNavOpen={null}
                 className={'some-class'}
             />,
         );
 
-        expect(wrapper.getElement()).toMatchSnapshot();
+        expect(renderResult.asFragment()).toMatchSnapshot();
+        expectMockedComponentPropsToMatchSnapshots([LeftNavHamburgerButton]);
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/expand-collapse-left-nav-hamburger-button.test.tsx]: removed

File: src/tests/unit/tests/common/components/__snapshots__/copy-issue-details-button.test.tsx.snap [hunk 10]
```diff
@@ -1,13 +1,20 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`CopyIssueDetailsButtonTest render 1`] = `
-"<Fragment>
-  <Toast deps={{...}} timeoutLength={6000} />
-  <CustomizedDefaultButton className="copy-issue-details-button" onClick={[Function (anonymous)]}>
-    <CopyIcon />
-    <div className="ms-Button-label">
+<DocumentFragment>
+  <div
```
[the remark this answers is not in the document]
Comment (v-sharmachir) [reply src/tests/unit/tests/common/components/__snapshots__/copy-issue-details-button.test.tsx.snap:5]: Hi @madalynrose, We already tried mocking the toast component but when we are unmocking it we are getting error.  And we need to unmock toast to compare the toast content. Thats why we keep it unmocked. Attaching Screenshots below.
Screenshot 2024-01-18 203610
Screenshot 2024-01-18 204121
Screenshot 2024-01-18 203623

**Review by madalynrose — APPROVED** [review]

**Pull request merged**